### PR TITLE
strongswan: add wolfssl plugin

### PR DESCRIPTION
**Pull request by lowjoel**
Maintainer: @pprindeville @Thermi
Compile tested: With gcc 7.5.0 on v19.07 x86_64-glibc and musl; gcc 11.2.0 on master fb7ff6b027d1c69e97e6d39e688a969c164065c8 aarch64 musl
Run tested: v19.07 x86_64 musl, gcc 7.5.0; v22.03rc1 aarch64-cortexa53 musl (Linksys E8400)

Description: Adds WolfSSL module to strongswan

This is a rebase of #14043. The initial code there seems to have worked but comments were left regarding the state of the version. Since there seems to be no longer activity from the original author, I've taken his branch and cherry-picked it on top of master. The PKG_RELEASE bump no longer seems necessary due to the use of $AUTORELEASE.

### CODE REVIEW

**Comment by lowjoel** [comment]
I'm not sure what the policy is regarding backports to release candidates; if this is suitable for inclusion into 22.03 too, I can open another PR targetting that branch once this is merged. Do let me know what you think.

**Comment by lowjoel** [comment]
Hey @pprindeville / @Thermi: gentle bump, not sure if you saw this. Could you have a look to see if this is suitable for merge? Guidance regarding what I should do for 22.03 would also be greatly appreciated.

(Thanks @\stintel for setting the right maintainers for this change.)

**Comment by lowjoel** [comment]
I've managed to build this on both master and 22.03. I've managed to install this on my hardware (running 22.03-rc1).

```
root@host:~#  swanctl --list-algs
encryption:
  AES_CTR[wolfssl]
  AES_CBC[wolfssl]
  3DES_CBC[wolfssl]
  DES_CBC[wolfssl]
  DES_ECB[wolfssl]
  NULL[wolfssl]
  AES_ECB[aes]
  RC2_CBC[rc2]
  AES_CFB[openssl]
  CAST_CBC[openssl]
  BLOWFISH_CBC[openssl]
integrity:
  HMAC_MD5_96[wolfssl]
  HMAC_MD5_128[wolfssl]
  HMAC_SHA1_96[wolfssl]
  HMAC_SHA1_128[wolfssl]
  HMAC_SHA1_160[wolfssl]
  HMAC_SHA2_256_128[wolfssl]
  HMAC_SHA2_256_256[wolfssl]
  HMAC_SHA2_384_192[wolfssl]
  HMAC_SHA2_384_384[wolfssl]
  HMAC_SHA2_512_256[wolfssl]
  HMAC_SHA2_512_512[wolfssl]
  AES_XCBC_96[xcbc]
aead:
  AES_GCM_16[wolfssl]
  AES_GCM_12[wolfssl]
  AES_CCM_16[wolfssl]
  AES_CCM_12[wolfssl]
  AES_CCM_8[wolfssl]
  CHACHA20_POLY1305[wolfssl]
  AES_GCM_8[openssl]
hasher:
  HASH_SHA1[wolfssl]
  HASH_MD5[wolfssl]
  HASH_SHA2_224[wolfssl]
  HASH_SHA2_256[wolfssl]
  HASH_SHA2_384[wolfssl]
  HASH_SHA2_512[wolfssl]
  HASH_SHA3_224[wolfssl]
  HASH_SHA3_256[wolfssl]
  HASH_SHA3_384[wolfssl]
  HASH_SHA3_512[wolfssl]
  HASH_MD4[openssl]
  HASH_IDENTITY[openssl]
prf:
  PRF_KEYED_SHA1[wolfssl]
  PRF_HMAC_MD5[wolfssl]
  PRF_HMAC_SHA1[wolfssl]
  PRF_HMAC_SHA2_256[wolfssl]
  PRF_HMAC_SHA2_384[wolfssl]
  PRF_HMAC_SHA2_512[wolfssl]
  PRF_FIPS_SHA1_160[fips-prf]
  PRF_AES128_XCBC[xcbc]
xof:
  XOF_SHAKE128[openssl]
  XOF_SHAKE256[openssl]
drbg:
dh:
  ECP_256[wolfssl]
  ECP_384[wolfssl]
  ECP_521[wolfssl]
  ECP_224[wolfssl]
  MODP_3072[wolfssl]
  MODP_4096[wolfssl]
  MODP_2048[wolfssl]
  MODP_2048_224[wolfssl]
  MODP_2048_256[wolfssl]
  MODP_1536[wolfssl]
  MODP_1024[wolfssl]
  MODP_1024_160[wolfssl]
  MODP_768[wolfssl]
  MODP_CUSTOM[wolfssl]
  CURVE_25519[wolfssl]
  MODP_6144[openssl]
  MODP_8192[openssl]
  CURVE_448[openssl]
  ECP_192[openssl]
  ECP_256_BP[openssl]
  ECP_384_BP[openssl]
  ECP_512_BP[openssl]
  ECP_224_BP[openssl]
rng:
  RNG_WEAK[wolfssl]
  RNG_STRONG[wolfssl]
  RNG_TRUE[random]
nonce-gen:
  NONCE_GEN[nonce]
```

**Comment by lowjoel** [comment]
Hey @pprindeville, gentle bump. Hope you can have this reviewed for inclusion soon.

**Comment by Neustradamus** [comment]
To follow this ticket

**Review by pprindeville — APPROVED** [review]
LGTM

**Pull request merged**
